test/pylib/scylla_cluster.py

```diff
@@ -740,7 +758,12 @@ def _write_config_file(self) -> None:
             with self.property_filename.open('w') as property_file:
                 for key, value in self.property_file.items():
                     property_file.write(f'{key}={value}\n')
-
+        if self.certificate_filename:
+            os.system('cp {} {}'.format(self.resources_certificate_file, self.certificate_filename))
```

Comment:
I see that tests use `os.system` as well. Does it really matter what crappy python library is used? Frankly, I like  `os.system` more since it is much more clear what it does without need to decipher python documentation.

Reply:
I see that `os.system` is used for other things but not copying files. I'd prefer if we stick to a single method of copying files across the codebase. And if something fails `shutil` will give a clearer error  -- it will throw an exception; if `cp` fails, I actually don't know if `os.system` will throw anything or just return an error code which  you don't check.

Reply:
so when we're inside Python let's try to write "Pythonic" code

Reply:
I got pretty good errors about missing file when I got path incorrectly. What is not pythonic about `os.system`?

Reply:
> What is not pythonic about os.system?

you're calling an external utility (`cp`) instead of using standard Python library. It's like you were writing in C++ and instead of using `std::filesystem::copy_file` you invoked `system("cp ...")`

Reply:
> > What is not pythonic about os.system?
> 
> you're calling an external utility (`cp`) instead of using standard Python library. It's like you were writing in C++ and instead of using `std::filesystem::copy_file` you invoked `system("cp ...")`

Python is a scripting language. I is fine to call external utility. We also use `os.system` to call `openssl` but python has a library to interface with it. Should we use it instead?

Reply:
> also `os.system` also introduces a shell. If you really wanted to use `cp` it would be preferable to use `subprocess` which doesn't introduce a shell. But why use such machinery when you can use standard library for copying files...

`subprocess`  interface in Python is a crime against humanity.

Reply:
> Python is a scripting language. I is fine to call external utility.

I don't see how the second statement follows from the first

> We also use os.system to call openssl but python has a library to interface with it. Should we use it instead?

If it's part of standard library then yes

Reply:
> > Python is a scripting language. I is fine to call external utility.
> 
> I don't see how the second statement follows from the first

They are literally the same. That what scripting language do. They glue  external utilities to achieve a task.

Reply:
Python can be used both as a scripting language and is a general-purpose language that can be used for implementing complex system, and that is what we're doing here with this test framework.

I can also glue external utilities in C++ if I want it so much.

Maybe you view Python as a scripting language only, and that it cannot be used for non-scripting purposes? I don't. Complex frameworks and apps are written in Python, see for example Django web framework

Reply:
> Python can be used both as a scripting language and is a general-purpose language 

Suppose so, though I wouldn't use if for anything that is larger than fits on a screen, but people write programs in brainfuck (https://gist.github.com/roachhd/dce54bec8ba55fb17d3a)  language so what do I know. So since it can be used for scripting why not use this capability? It is like saying that C++ can be used to write in object oriented way  and functional way so lets use only one of them.

Reply:
Replaced with god given native Python library. No more pesky shells.